client-src/elements/chromedash-all-features-page.ts

```diff
@@ -59,28 +69,28 @@ export class ChromedashAllFeaturesPage extends LitElement {
     }
 
     if (this.rawQuery.hasOwnProperty('q')) {
-      this.query = this.rawQuery['q'];
+      this.query = this.rawQuery['q'] ?? this.query;
```

Comment:
I'd use `!` here instead of implying that the field could be null. Or Typescript should understand that this pattern guarantees non-null-ness:
```suggestion
    if (this.rawQuery.q != null) {
      this.query = this.rawQuery.q;
```
I'm not sure why the original code used `hasOwnProperty`; maybe an old worry about prototype pollution.